src/Facebook/InstantArticles/Transformer/Transformer.php

```diff
@@ -57,6 +57,43 @@ class Transformer
      */
     const INSTANT_ARTICLES_PARSED_FLAG = 'data-instant-articles-element-processed';
 
+    /**
+     * LOG_NOLOG means no log will be generated during transformation.
+     */
+    const LOG_OFF = 'OFF';
+
+    /**
+     * LOG_INFO **default** Just the informative logs will be turned on.
+     */
+    const LOG_INFO = 'INFO';
+
+    /**
+     * LOG_DEBUG means all verbose information will be active for debugging purposes.
+     */
+    const LOG_DEBUG = 'DEBUG';
+
+    /**
+     * LOG_ERROR means only errors will be shown.
+     */
+    const LOG_ERROR = 'ERROR';
+
+    const LOG_DEFINITION = array(
+        self::LOG_OFF => 0,
+        self::LOG_ERROR => 1,
+        self::LOG_INFO => 2,
+        self::LOG_DEBUG => 3
+    );
+
+    /**
+     * @var string $logLevel The log level set into this transformer instance. Possible values: LOG_NOLOG, LOG_INFO or LOG_DEBUG
+     */
+    private $logLevel = self::LOG_INFO;
+
+    /**
+     * @var string[] $logs The log messages generated during the transformation process.
+     */
+    private $logs = ['Possible log levels: OFF, ERROR, INFO or DEBUG'];
```

Comment:
How about WARN or FATAL?

Reply:
Warnings are all ready to be accessed from `$transformer->getWarnings()`.

Fatals I see as the Exception handling being the best approach.